includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -225,10 +225,13 @@ public function create_payment_intent( $order_id = null ) {
 			return $this->create_setup_intent();
 		}
 
+		$enabled_payment_methods = array_filter( $this->get_upe_enabled_payment_method_ids(), [ $this, 'is_enabled_at_checkout' ] );
+		$enabled_payment_methods = array_filter( $enabled_payment_methods, [ $this, 'is_enabled_for_site_currency' ] );
```

Comment:
There is a actually a small reason to keep these functions separated, but I think I've found a nice workaround. Essentially, when `is_enabled_at_checkout` is called from inside `create_payment_intent_ajax` or `update_payment_intent_ajax`, we know that this is a `wc_ajax` call and `woocommerce_get_currency` will return the customer's assigned currency. However, when called inside `get_enabled_payment_method_config` or when `is_enabled_for_saved_payments` is called inside `payment_fields`, these functions are called within the admin context, which means `get_woocommerce_currency` returns the store's default currency instead. 

If the customer, for example, has chosen to use EUR, whereas the store, while supporting multi-currency, has USD as its default currency, we will be unnecessarily limiting the `config` data returned inside `payment_fields` and this could cause some bad things to happen.

To resolve this, I've added an admin check on the filter.
```
		return $this->payment_methods[ $payment_method_id ]->is_enabled_at_checkout()
			&& ( is_admin() || $this->payment_methods[ $payment_method_id ]->is_currency_valid() );
```
In this case, if the request is made in admin context, the currency filter is ignored. I think this keeps things tidy, as you suggested, without any interference!

Reply:
Ah gotcha, I thought those were two different functions (getting payment methods for admin use and getting payment methods for customer use). What you came up with here works for me!